cql3/statements/select_statement.cc

```diff
@@ -1400,12 +1401,26 @@ indexed_table_select_statement::find_index_partition_ranges(query_processor& qp,
     using value_type = std::tuple<dht::partition_range_vector, lw_shared_ptr<const service::pager::paging_state>>;
     auto now = gc_clock::now();
     auto timeout = db::timeout_clock::now() + get_timeout(state.get_client_state(), options);
+    bool is_paged = options.get_page_size() >= 0;
+    constexpr size_t MAX_PARTITION_RANGES = 1000;
+    // This check ensures that `partition_range_vector` won't cause a large allocation (see #18536).
+    // This is not a fatal error, but we treat it as such to ensure this won't go under the radar.
+    // If this fails, please adjust MAX_PARTITION_RANGES accordingly.
+    if(MAX_PARTITION_RANGES * sizeof(dht::partition_range_vector::value_type) > current_allocator().preferred_max_contiguous_allocation()) {
+        on_fatal_internal_error(logger, seastar::format(
```

Comment:
I thought a static_assert would work. A runtime fatal error is too much, we can just emit a warning.